components/dot_menu/dot_menu.tsx

```diff
@@ -26,73 +27,79 @@ import DotsHorizontalIcon from 'components/widgets/icons/dots_horizontal';
-        commentCount: PropTypes.number,
-        isFlagged: PropTypes.bool,
-        handleCommentClick: PropTypes.func,
-        handleDropdownOpened: PropTypes.func,
-        handleAddReactionClick: PropTypes.func,
-        isMenuOpen: PropTypes.bool,
-        isReadOnly: PropTypes.bool,
-        pluginMenuItems: PropTypes.arrayOf(PropTypes.object),
-        isLicensed: PropTypes.bool, // TechDebt: Made non-mandatory while converting to typescript
-        postEditTimeLimit: PropTypes.string, // TechDebt: Made non-mandatory while converting to typescript
-        enableEmojiPicker: PropTypes.bool.isRequired,
-        channelIsArchived: PropTypes.bool, // TechDebt: Made non-mandatory while converting to typescript
-        currentTeamUrl: PropTypes.string, // TechDebt: Made non-mandatory while converting to typescript
-
-        /*
-         * Components for overriding provided by plugins
+        /**
+         * Function to unflag the post
          */
-        components: PropTypes.object, // TechDebt: Made non-mandatory while converting to typescript
-
-        actions: PropTypes.shape({
-
-            /**
-             * Function flag the post
-             */
-            flagPost: PropTypes.func.isRequired,
-
-            /**
-             * Function to unflag the post
-             */
-            unflagPost: PropTypes.func.isRequired,
-
-            /**
-             * Function to set the editing post
-             */
-            setEditingPost: PropTypes.func.isRequired,
-
-            /**
-             * Function to pin the post
-             */
-            pinPost: PropTypes.func.isRequired,
-
-            /**
-             * Function to unpin the post
-             */
-            unpinPost: PropTypes.func.isRequired,
-
-            /**
-             * Function to open a modal
-             */
-            openModal: PropTypes.func.isRequired,
-
-            /*
-             * Function to set the unread mark at given post
-             */
-            markPostAsUnread: PropTypes.func.isRequired,
-        }), // TechDebt: Made non-mandatory while converting to typescript
-
-        canEdit: PropTypes.bool, // TechDebt: Made non-mandatory while converting to typescript
-        canDelete: PropTypes.bool, // TechDebt: Made non-mandatory while converting to typescript
-    }
+        unflagPost: (postId: string) => void;
+
+        /**
+         * Function to set the editing post
+         */
+        setEditingPost: (postId?: string, commentCount?: number, refocusId?: string, title?: string, isRHS?: boolean) => void;
+
+        /**
+         * Function to pin the post
+         */
+        pinPost: (postId: string) => void;
+
+        /**
+         * Function to unpin the post
+         */
+        unpinPost: (postId: string) => void;
+
+        /**
+         * Function to open a modal
+         */
+        openModal: (postId: any) => void;
```

Comment:
Can this be made `string`?

Reply:
there are invocations that are not `string`. Leaving as is for now.